src/main/java/com/sequenceiq/cloudbreak/controller/validation/UpdateStackRequestValidator.java

```diff
@@ -14,11 +14,13 @@ public void initialize(ValidUpdateStackRequest constraintAnnotation) {
 
     @Override
     public boolean isValid(UpdateStackJson value, ConstraintValidatorContext context) {
-        if (value.getStatus() != null && value.getHostGroupAdjustment() != null) {
-            addConstraintViolation(context, "Invalid PUT request on this resource. NodeCount and status cannot be set in the same request.");
+        if (value.getStatus() != null && value.getHostGroupAdjustment() != null && value.getAllowedSubnets() != null) {
+            addConstraintViolation(context, "Invalid PUT request on this resource. "
+                    + "Node count, status and security constraints cannot be set in the same request.");
```

Comment:
This validator doesn't validate if 2 different property is set in the update request (e.g.: status & hostGroupAdjustment) only if all 3 are set, but the Controller handles only one update property.